Core/Core/Dashboard/View/DashboardCardView.swift

```diff
@@ -103,14 +106,39 @@ public struct DashboardCardView: View {
     }
 
     @ViewBuilder
-    private var layoutToggleButton: some View {
-        if cards.shouldShowLayoutToggleButton {
-            Button(action: layoutViewModel.toggle) {
-                layoutViewModel.buttonImage
+    private var settingsButton: some View {
+        if cards.shouldShowSettingsButton {
+            Button {
+                guard controller.value.presentedViewController == nil else {
+                    controller.value.presentedViewController?.dismiss(animated: true)
+                    return
+                }
+                let interactor = DashboardSettingsInteractorLive(environment: env, defaults: env.userDefaults!)
+                let viewModel = DashboardSettingsViewModel(interactor: interactor)
+                let dashboard = CoreHostingController(DashboardSettingsView(viewModel: viewModel))
+                dashboard.addDoneButton(side: .left)
+                let container = HelmNavigationController(rootViewController: dashboard)
+                container.preferredContentSize = viewModel.popoverSize
+                container.modalPresentationStyle = .popover
+
+                // Position the popover's arrow to point to the settings button
+                if let popoverController = container.popoverPresentationController {
+                    popoverController.sourceView = controller.value.navigationItem.rightBarButtonItem?.customView
+                    popoverController.sourceRect = CGRect(x: 26, y: 35, width: 0, height: 0)
+                }
+
+                env.router.show(
+                    container,
+                    from: controller,
+                    options: .modal(.popover),
+                    analyticsRoute: "/dashboard/settings"
+                )
+            } label: {
```

Comment:
I think creating view controllers, view models, interactors are out of the scope of the view's responsibility. A view should signal to its' view model that a button has been pressed and the view model should delegate the creation/routing further up.